jira_integration_example/utilities/jira_integration.py

```diff
@@ -67,15 +72,26 @@ def update_jira_based_on_monitoring_notification(jira_client, jira_project,
     except KeyError as e:
         raise NotificationParseError(f"Notification is missing required dict key: {str(e)}")
 
+    incident_id_label = f'monitoring_incident_id_{incident_id}'
+
     if incident_state == 'open':
         summary = '%s - %s' % (incident_condition_name, incident_resource_name)
         description = '%s\nSee: %s' % (incident_summary, incident_url)
         issue = jira_client.create_issue(
             project=jira_project,
             summary=summary,
             description=description,
-            issuetype={'name': 'Bug'})
+            issuetype={'name': 'Bug'},
+            labels=[incident_id_label])
         logger.info('Created jira issue %s', issue)
-    elif incident_state != 'closed':
+
+    elif incident_state == 'closed':
+        incident_issues = jira_client.search_issues(
+            f'labels = {incident_id_label} AND status != {jira_status}')
+        for issue in incident_issues:
```

Comment:
You should probably log a warning here if no issues are found to close. Will make it easier to know how often this is failing, or how often people are manually closing issues instead of letting this close issues for them.

Reply:
Done